### ISSUE TITLE
payload.attachments is always nil in handleNotificationReceived and handleNotificationAction

### ISSUE THREAD

**Issue by 6franek** [OneSignal/OneSignal-iOS-SDK#239]
I would like to get `payload.attachments` in `handleNotificationReceived`, so that I can save media attachments in the app when the notification is received. The problem is that `payload.attachments` is always nil in `handleNotificationReceived` and also in `handleNotificationAction`. All the other data in the payload are correct. I can see the sent image in my iPhone's notifications center, but I can't obtain its path it anywhere in the code.

This is how I initialize and configure the SDK
```
OneSignal.initWithLaunchOptions(launchOptions, appId: appId, handleNotificationReceived: {
            (notification) in
            if let notification = notification {
                let id = notification.payload.notificationID
                let url = notification.payload.launchURL
                let title = notification.payload.title
                let body = notification.payload.body
                var imagePath: String? = nil
                let attachments = notification.payload.attachments
                if let attachments = notification.payload.attachments {
                    imagePath = attachments!["id"] as? String
                }
                var validUntil: Int? = nil
                if let additionalData = notification.payload.additionalData {
                    validUntil = additionalData["validUntil"] as? Int
                }
                (...)
            }
        }, handleNotificationAction: {
            (result) in
            if let result = result {
                if let additionalData = result.notification.payload.additionalData {
                    print("additional data: \(additionalData)")
                }
                if let attachments = result.notification.payload.attachments {
                    print("attachments: \(attachments)")
                }
            }
        }, settings: [kOSSettingsKeyInAppLaunchURL: false, kOSSettingsKeyInFocusDisplayOption: OSNotificationDisplayType.none.rawValue])
```

**Comment by noelportugal** [comment]
having the same issue! I see the ios_attachment in rawPayload, but attachments comes back nil
[AnyHashable("aps"): {
    alert =     {
        body = "Test Body";
        title = "Test Title";
    };
    "content-available" = 1;
    "mutable-content" = 1;
    sound = "inquisitiveness.m4r";
}, AnyHashable("att"): {
    id = "https://media.giphy.com/media/l0Iy9g9pvspPyJIju/giphy.gif";
}, AnyHashable("custom"): {
    i = "a872d8e7-e7e7-4eee-9e78-bca1b942a8ad";
}]

**Comment by Nightsd01** [comment]
We recently fixed some parsing issues with the notification payload. After doing some testing, this issue appears to be resolved as of the current version 2.6.1.

Please feel free to comment on this issue if you (or anyone else) encounters this issue again.
